beams/sequencer/helpers/Worker.py

```diff
@@ -6,42 +6,56 @@
 """
 import logging
 from multiprocessing import Process, Value
+from typing import Callable, Any, Optional, Union, List
 
+from epics.multiproc import CAProcess
 
-class Worker:
-    def __init__(self, proc_name, stop_func=None, work_func=None):
-        self.do_work = Value("i", False)
+
+class Worker():
+    def __init__(self, 
+                 proc_name: str, 
+                 stop_func: Optional[Callable[[None], None]] = None, 
+                 work_func: Optional[Callable[[Any], None]] = None,
+                 proc_type: Union[Process, CAProcess] = Process,
```

Comment:
Tiny tiny type hinting nitpicks:
- CAProcess is a subclass of Process, so the Union is unnecessary for the type hint
- We pass in a type here rather than an object, so the type keyword is needed

```suggestion
                 proc_type: type[Process] = Process,
```

Reply:
Sweet, ya I  was slightly abusing the notation to show the users what types of things to pass here but from an OOP standpoint and generally you are totally correct.